x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -53,11 +76,85 @@ public class IlmHealthIndicatorService implements HealthIndicatorService {
     );
 
     public static final String AUTOMATION_DISABLED_IMPACT_ID = "automation_disabled";
+    public static final String STAGNATING_INDEX_IMPACT_ID = "stagnating_index";
+    public static final List<HealthIndicatorImpact> AUTOMATION_DISABLED_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            AUTOMATION_DISABLED_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management is disabled. The performance and stability of the cluster "
+                + "could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    public static final List<HealthIndicatorImpact> STAGNATING_INDEX_IMPACT = List.of(
+        new HealthIndicatorImpact(
+            NAME,
+            STAGNATING_INDEX_IMPACT_ID,
+            3,
+            "Automatic index lifecycle and data retention management cannot make progress on one or more indices. The performance and "
+                + "stability of the indices and/or the cluster could be impacted.",
+            List.of(ImpactArea.DEPLOYMENT_MANAGEMENT)
+        )
+    );
+
+    private static final TimeValue ONE_DAY = TimeValue.timeValueDays(1);
```

Comment:
Shall we have this configurable via a setting?

Similar to the ones we use in the coordination diagnostics service - https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsService.java#L138 
and document the new one [here](https://www.elastic.co/guide/en/elasticsearch/reference/8.8/health-diagnostic-settings.html)

We should still default to 24 hours, but it'll give us a lever to make the health API stop returning YELLOW if we need to.

Reply:
sounds cool. Will do it. btw the documentation will be in a different PR

Reply:
well, all these rules are statically created, so there's no access to the `ClusterService` at that point, implying deeper changes. I'd rather do it in a different PR because this one is bloated already

Reply:
Sounds good, thanks Pablo